File: September/September_14/230B.py
import math
def Tprime(n):
    count=0
    for i in range(2,int(math.sqrt(n)+1)):
        if n%i==0:
            count+=1
            if count>1:
                break
    if count>1 or count==0:
        return "NO"
    else:
        return "YES"

n = int(input())
a = list(map(int,input().split()))
for i in range(n):
    ans = Tprime(a[i])
    print(ans)

File: JULY/JULY_17/230B.cpp
#include<iostream>
using namespace std;

int main(){
	long long int n;
	cin >> n;
	long long int a[n];
	for(long long int i=0; i<n; i++) cin >> a[i];
	for (long long int i=0; i<n; i++) {
		int count = 0, flag=0;
		for (long long int j=1; j<=a[i]; j++){
			if(a[i]%j==0) count++;
			if(count>3) {
				flag=1;
				break;
			}
		}
		if(flag==1 || count<3 ) cout << "NO" << endl;
		else 		cout << "YES" << endl;
	}
	return 0;
}
